src/util/secure_socket_thread.c

```diff
@@ -0,0 +1,214 @@
+#include <fcntl.h>
+#include <kmyth/formatting_tools.h>
+
+#include "charbuf.h"
+#include "pelz_log.h"
+#include "pelz_socket.h"
+#include "pelz_json_parser.h"
+#include "pelz_io.h"
+#include "pelz_service.h"
+#include "pelz_request_handler.h"
+#include "secure_socket_thread.h"
+
+#include "sgx_urts.h"
+#include "pelz_enclave.h"
+#include ENCLAVE_HEADER_UNTRUSTED
+
+#define BUFSIZE 1024
+#define MODE 0600
+
+static void *secure_process_wrapper(void *arg)
+{
+  secure_socket_process(arg);
+  pthread_exit(NULL);
+}
+
+void secure_socket_thread(void *arg)
+{
+  ThreadArgs *threadArgs = (ThreadArgs *) arg;
+  int socket_listen_id = threadArgs->socket_id;
+  int max_requests = threadArgs->max_requests;
+  pthread_mutex_t lock = threadArgs->lock;
+
+  ThreadArgs processArgs;
+  pthread_t stid[max_requests];
+  int socket_id = 0;
+
+  do
+  {
+    if (pelz_key_socket_accept(socket_listen_id, &socket_id))
+    {
+      pelz_log(LOG_ERR, "Socket Client Connection Error");
+      continue;
+    }
+
+    if (socket_id == 0)         //This is to reset the while loop if select() times out
+    {
+      continue;
+    }
+
+    if (socket_id > max_requests)
+    {
+      pelz_log(LOG_WARNING, "%d::Over max socket requests.", socket_id);
+      pelz_key_socket_close(socket_id);
+      continue;
+    }
+
+    processArgs.lock = lock;
+    processArgs.socket_id = socket_id;
+    if (pthread_create(&stid[socket_id], NULL, secure_process_wrapper, &processArgs) != 0)
+    {
+      pelz_log(LOG_WARNING, "%d::Failed to create thread.", socket_id);
+      pelz_key_socket_close(socket_id);
+      continue;
+    }
+
+    pelz_log(LOG_INFO, "Secure Socket Thread %d, %d", (int) stid[socket_id], socket_id);
+  }
+  while (socket_listen_id >= 0 && socket_id <= (max_requests + 1));
+  return;
+}
+
+//This function will need to be changed with the attestation handshake and process flow
+void secure_socket_process(void *arg)
+{
+  ThreadArgs *processArgs = (ThreadArgs *) arg;
+  int new_socket = processArgs->socket_id;
+  pthread_mutex_t lock = processArgs->lock;
+
+  charbuf request;
+  charbuf message;
+  RequestResponseStatus status;
+  const char *err_message;
+
+  //Attestation handshake function should be added here in the process flow
+
+  while (!pelz_key_socket_check(new_socket))
+  {
+    //Receiving request and Error Checking
+    if (pelz_key_socket_recv(new_socket, &request))
+    {
+      pelz_log(LOG_ERR, "%d::Error Receiving Request", new_socket);
+      while (!pelz_key_socket_check(new_socket))
+      {
+        continue;
+      }
+      pelz_key_socket_close(new_socket);
+      return;
+    }
+
+    pelz_log(LOG_DEBUG, "%d::Request & Length: %.*s, %d", new_socket, (int) request.len, request.chars, (int) request.len);
+
+    RequestType request_type = REQ_UNK;
+
+    charbuf key_id;
+    charbuf data_in;        
+    charbuf data_out;       
+    charbuf request_sig;
+    charbuf requestor_cert;
+
+    charbuf data;           //Remove this variable
```

Comment:
Am I correct thinking the comment relates to changes that will have to happen once we roll in the attestation code?

Reply:
This was commenting recommended changes for the attestation code when implemented but was not need after further planning of how we would handle the encrypted portions of the requests.